### PR TITLE
feat(es/lints): Implement no-dupe-args

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

Implement [no-dupe-args](https://eslint.org/docs/rules/no-dupe-args)

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_ecma_lints

**Comment by magic-akari** [comment]
It should probably be set as a critical error.
In strict mode, all naming duplicates should be considered as errors.
In non-strict mode, named duplicates are only allowed in normal functions, while errors will be reported in arrow functions and class methods.

What do you think about it? @kdy1

**Comment by kdy1** [comment]
Yeah I agree that it should be a critical error by default.

File: crates/swc_ecma_lints/src/rules/no_dupe_args.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        });
+    }
+}
+
+impl Visit for NoDupeArgs {
+    noop_visit_type!();
+
+    fn visit_function(&mut self, f: &Function) {
+        let variables: Vec<(JsWord, Span)> = find_ids(&f.params);
+
```
Comment (kdy1) [inline crates/swc_ecma_lints/src/rules/no_dupe_args.rs:44]: I think we should visit children

File: crates/swc_ecma_lints/src/rules/no_dupe_args.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    fn visit_arrow_expr(&mut self, arrow_fn: &ArrowExpr) {
+        let variables: Vec<(JsWord, Span)> = find_ids(&arrow_fn.params);
+
+        self.check(variables);
+    }
+
+    fn visit_constructor(&mut self, n: &Constructor) {
+        let variables: Vec<(JsWord, Span)> = find_ids(&n.params);
+
```
Comment (kdy1) [inline crates/swc_ecma_lints/src/rules/no_dupe_args.rs:56]: Same here

File: crates/swc_ecma_lints/src/rules/no_dupe_args.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    noop_visit_type!();
+
+    fn visit_function(&mut self, f: &Function) {
+        let variables: Vec<(JsWord, Span)> = find_ids(&f.params);
+
+        self.check(variables);
+    }
+
+    fn visit_arrow_expr(&mut self, arrow_fn: &ArrowExpr) {
+        let variables: Vec<(JsWord, Span)> = find_ids(&arrow_fn.params);
```
Comment (kdy1) [inline crates/swc_ecma_lints/src/rules/no_dupe_args.rs:49]: Same here

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_lints

**Pull request merged**